### PR TITLE
Disable csrf as it is not needed any more

### PR DESCRIPTION
**Pull request by krusche**
Fixes https://github.com/ls1intum/Artemis/issues/1421 

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.

### Motivation and Context
CSRF is not supported correctly in Safari and is not needed any more. We do not use server sessions for authentication. With JWT, cross-site attacks are not really possible any more. Here are some further links with additional information:

* https://scotthelme.co.uk/csrf-is-dead/
* https://security.stackexchange.com/questions/170388/do-i-need-csrf-token-if-im-using-bearer-jwt

### Steps for Testing
1. Logout and Log in to Artemis
2. Try out different actions, including creating new exercises and deleting those
3. Try out different browsers, in particular Safari

**New Tests:**
1. Embed external images (from other urls) in Markdown and check that they can still be viewed
2. Add lecture video units from Youtube and check that they can still be watched
3. Open the developer tools and the network tab: check the header values of the REST responses and make sure those values are correct. Compare them with the values set in the SecurityConfiguration (see Files changed)

Open the network tab, and check that no 403 errors occur for POST and DELETE requests. Also make sure that nothing else is broken. Note, that we did not change any functionality here, so if specific functionality is broken, please compare it with `develop` and potentially create a Github issue as it is most probably not related to the changes in this PR.

### Test Coverage
no change

### Screenshots
no UI changes

### CODE REVIEW

**Review by tiffca — APPROVED** [review]
tested successfully, no issues encountered

**Review by airelawaleria — APPROVED** [review]

**Review by JohannesWeiss — APPROVED** [review]
Tested with a few exercises, did not encounter any problems. Using Safari 14.

**Review by EneaGore — APPROVED** [review]

**Review by marvinchimeka — DISMISSED** [review]

**Review by simonzilker — DISMISSED** [review]

**Comment by jpbernius** [comment]
With the new CSP Headers, we need to test: downloading lecture attachments, downloading file uploads, embedded videos, and external images in markdown.

Note: On our testservers, the CSP header is currently overwritten by our nginx proxy server, so we need to update the config here before testing.

**Review by FrankeLukas — CHANGES_REQUESTED** [review]
When I embed a YouTube video on TS2 using Firefox I get `Content Security Policy: Die Einstellungen der Seite haben das Laden einer Ressource auf https://www.youtube.com/embed/p-m0XnRHWMo blockiert ("frame-src").` when trying to view the Video Unit.

**Comment by tiffca** [comment]
on ts3 the video in lecture units didn't work for me. 

Console Error: 
Refused to frame 'https://www.youtube.com/' because it violates the following Content Security Policy directive: "frame-src 'self' data:".
On Chrome (it doesn't show on Safari either): 
image

configuration
image

**Review by JohannesWeiss — COMMENTED** [review]
Can confirm the issue in Safari, too. There, nothing except the error in the console is shown.

File: src/main/java/de/tum/in/www1/artemis/config/SecurityConfiguration.java [hunk 1]
```diff
@@ -125,16 +124,20 @@ protected void configure(HttpSecurity http) throws Exception {
...
-        .and()
+            .disable()
             .addFilterBefore(corsFilter, UsernamePasswordAuthenticationFilter.class)
-            .exceptionHandling().authenticationEntryPoint(problemSupport)
-            .accessDeniedHandler(problemSupport)
+                .exceptionHandling().authenticationEntryPoint(problemSupport)
+                .accessDeniedHandler(problemSupport)
         .and()
             .headers()
+            .contentSecurityPolicy("default-src 'self'; frame-src 'self' data:; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; img-src * data:; font-src 'self' data:")
```
Comment (jpbernius) [inline src/main/java/de/tum/in/www1/artemis/config/SecurityConfiguration.java:133]: Extend policy to whitelist youtube:
```suggestion
            .contentSecurityPolicy("default-src 'self'; frame-src 'self' data: https://www.youtube-nocookie.com/embed/ https://www.youtube.com/embed/ https://youtu.be/embed/ https://youtube-nocookie.com/embed/ https://youtube.com/embed/; script-src 'self' 'unsafe-inline' 'unsafe-eval' https://www.youtube.com/iframe_api https://*.ytimg.com/; style-src 'self' 'unsafe-inline' https://*.ytimg.com/; img-src * data: https://*.youtube-nocookie.com https://*.youtube.com https://*.ytimg.com/ ; font-src 'self' data:; child-src 'self' https://www.youtube-nocookie.com/ https://www.youtube.com/ https://youtu.be/ https://youtube-nocookie.com/ https://youtube.com/; object-src 'self' https://*.youtube-nocookie.com https://*.youtube.com; connect-src 'self' https://www.youtube-nocookie.com/ https://www.youtube.com/ https://youtube-nocookie.com/ https://youtube.com/; media-src 'self' https://*.youtube-nocookie.com https://*.youtube.com")
```

Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/config/SecurityConfiguration.java:133]: The problem is that instructors could also decide to use videos on other platforms, e.g. Vimeo or even custom ones from their university, so we should probably allow anything here

**Comment by krusche** [comment]
We simplified the Content Security Policy for now and added a TODO to make it more safe in the future

**Review by jpbernius — APPROVED** [review]

**Pull request merged**